timeout.go

```diff
@@ -0,0 +1,41 @@
+package codacytool
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"time"
+)
+
+func timeoutSeconds() time.Duration {
+	value, exists := os.LookupEnv("TIMEOUT_SECONDS")
+
+	if !exists {
+		return defaultTimeout
+	}
+
+	seconds, err := strconv.Atoi(value)
+	if err != nil {
+		return defaultTimeout
+	}
+
+	return time.Duration(seconds) * time.Second
+}
+
+func runMethodWithTimeout(method func(), timeoutExceeded func(), maxDuration time.Duration) {
+	ctx := context.Background()
+
+	c1 := make(chan struct{})
```

Comment:
Instead of doing this, I replaced the `chan struct{}` with `chan toolRunResult`, which is a struct that contains the tool result. When the tool finishes running, the result is sent over this channel and the success handler prints the result.
By doing this way, `startToolImplementation` returns the result instead of printing it after calling the tool, which is also a lot better to test

What do you think about this solution?